internal/tweets/tweets.go

```diff
@@ -224,16 +225,26 @@ func UnlikeTweet(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 func Retweet(w http.ResponseWriter, r *http.Request) {
-	tweetID := r.FormValue("tweet_id")
-	if tweetID == "" {
-		http.Error(w, "Missing tweet ID", http.StatusBadRequest)
+	tweetID := mux.Vars(r)["id_tweet"]
+	userID, ok := r.Context().Value("userID").(int)
+	if !ok {
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
 		return
 	}
 
-	userID, err := services.GetCurrentUserID(r)
-
-	query := "SELECT COUNT(*) FROM retweets WHERE user_id = $1 AND tweet_id = $2"
+	query := "SELECT COUNT(*) FROM tweets WHERE tweet_id = $1"
```

Comment:
this count can be omitted/replaced with https://github.com/pm888/Twitter/pull/15/files#diff-021101cb4f49e714e7925204ff92ef37b8526e298efcbf7296a64942d58651e0R258